### ISSUE TITLE
Corfu autocompletion not working with default config and template

### ISSUE THREAD

**Issue by aadi58002** [minad/tempel#93]
Hello there,
# Expected
Get the templates which can be inserted in the current mode be present in the completion list.
# Actually happens
Don't get any completion from the templates

Config
```elisp
  ;; Configure Tempel
  (use-package tempel
    ;; Require trigger prefix before template name when completing.
    ;; :custom
    ;; (tempel-trigger-prefix "<")

    :bind (("M-+" . tempel-complete) ;; Alternative tempel-expand
           ("M-*" . tempel-insert))

    :init

    ;; Setup completion at point
    (defun tempel-setup-capf ()
      ;; Add the Tempel Capf to `completion-at-point-functions'.
      ;; `tempel-expand' only triggers on exact matches. Alternatively use
      ;; `tempel-complete' if you want to see all matches, but then you
      ;; should also configure `tempel-trigger-prefix', such that Tempel
      ;; does not trigger too often when you don't expect it. NOTE: We add
      ;; `tempel-expand' *before* the main programming mode Capf, such
      ;; that it will be tried first.
      (setq-local completion-at-point-functions
                  (cons #'tempel-expand
                        completion-at-point-functions)))

    (add-hook 'prog-mode-hook 'tempel-setup-capf)
    (add-hook 'text-mode-hook 'tempel-setup-capf)
    (global-tempel-abbrev-mode)
  )
(use-package tempel-collection)
(use-package corfu
  :init
  (global-corfu-mode)
  ;; Setup corfu for popup like completion
  (setq corfu-cycle t  ; Allows cycling through candidates
        corfu-auto t   ; Enable auto completion
        corfu-auto-prefix 1  ; Complete with less prefix keys
        corfu-auto-delay 0.0  ; No delay for completion
        corfu-echo-documentation 0.25  ; Echo docs for current completion option
        corfu-quit-at-boundary 'insert
        )
  (global-corfu-mode 1)
  (advice-add #'lsp-completion-at-point :around #'cape-wrap-noninterruptible))
(use-package emacs
  :init
  (defun crm-indicator (args)
    (cons (format "[CRM%s] %s"
                  (replace-regexp-in-string
                   "\\`\\[.*?]\\*\\|\\[.*?]\\*\\'" ""
                   crm-separator)
                  (car args))
          (cdr args)))
  (advice-add #'completing-read-multiple :filter-args #'crm-indicator)

  (setq minibuffer-prompt-properties
        '(read-only t cursor-intangible t face minibuffer-prompt))
  (add-hook 'minibuffer-setup-hook #'cursor-intangible-mode)
  (setq enable-recursive-minibuffers t)
  (setq completion-cycle-threshold 3)
  (setq tab-always-indent 'complete))
(use-package cape
    :init
    (add-to-list 'completion-at-point-functions #'cape-file)
    (add-to-list 'completion-at-point-functions #'cape-dabbrev))
```

I checked the variables `completion-at-point` and it does have the tempel-expand in the list
![Screenshot_20230214_101059](https://user-images.githubusercontent.com/96418027/218641058-c3d64f85-cb53-436d-ae40-ed93804b00fe.png)

**Comment by minad** [comment]
The completin at point functions are tried in order. Therefore tempel  will never get to run if it comes late in the list. tempel-expand only expands full template names manually.

**Comment by zetashift** [comment]
> The completin at point functions are tried in order. Therefore tempel will never get to run if it comes late in the list. tempel-expand only expands full template names manually. Note that you can also press `M-+` or `M-*`.

As a complete emacs noob, what is the way to get tempel early up in the list? I'm using `tempel` and `corfu`

**Comment by aadi58002** [comment]
> > The completin at point functions are tried in order. Therefore tempel will never get to run if it comes late in the list. tempel-expand only expands full template names manually. Note that you can also press `M-+` or `M-*`.
> 
> As a complete emacs noob, what is the way to get tempel early up in the list? I'm using `tempel` and `corfu`

@minad Can you address the above question? I have the same question

**Comment by minad** [comment]
These are basic questions. Check out the Emacs and Elisp manuals. :)

**Comment by zetashift** [comment]
> These are basic questions. Check out the Emacs and Elisp manuals. :)

I ended up just relying on a keybinding to call `tempel-expand`. I tried looking through the emacs manuals (my elisp-fu really sucks) about completion in buffers, but it's all quite new to me, so I couldn't get it working. There was also a recent [reddit thread[(https://www.reddit.com/r/emacs/comments/116x4nq/completion_at_point_how_to_keep_tempelexpand_at/) about it, but the answer made me have even more questions.

Currently Tempel works great for my needs, so thank you! And thank you for all your other packages as well, they are awesome :D.
